src/EnergyPlus/FanCoilUnits.cc

```diff
@@ -1938,13 +1941,14 @@ namespace FanCoilUnits {
 			}
 			// obtain unit output with no active heating/cooling
 			Calc4PipeFanCoil( FanCoilNum, ControlledZoneNum, FirstHVACIteration, QUnitOutNoHC, 0.0 );
-			if ( FirstHVACIteration ) {
-				FanCoil( FanCoilNum ).QUnitOutNoHC = QUnitOutNoHC;
-			}
-			else {
+
+			if ( ColdFlowLocked || HotFlowLocked ) {
 				QUnitOutNoHC = FanCoil( FanCoilNum ).QUnitOutNoHC;
+			} else {  // continue to update QUnitOutNoHC while flow is unlocked
+				FanCoil( FanCoilNum ).QUnitOutNoHC = QUnitOutNoHC;
```

Comment:
This is the simple fix. Continue updating QUnitOutNoHC until flow is locked (instead of only on FirstHVACIteration). The only problem I see is if the plant resolver locks the flow at something other than what it was the previous iteration (starved flow, etc).